sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -163,6 +168,38 @@ public String getNonProxyHosts() {
      * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE}.
      */
     public static ProxyOptions fromConfiguration(Configuration configuration) {
+        return fromConfiguration(configuration, true);
+    }
+
+    /**
+     * Attempts to load a proxy from the environment.
+     * <p>
+     * If a proxy is found and loaded, the proxy address is DNS resolved based on {@code resolveProxy}. When {@code
+     * resolveProxy} is false resolving it before usage may be required.
+     * <p>
+     * Environment configurations are loaded in this order:
+     * <ol>
+     *     <li>Azure HTTPS</li>
+     *     <li>Azure HTTP</li>
+     *     <li>Java HTTPS</li>
+     *     <li>Java HTTP</li>
+     * </ol>
+     *
+     * Azure proxy configurations will be preferred over Java proxy configurations as they are more closely scoped to
+     * the purpose of the SDK. Additionally, more secure protocols, HTTPS vs HTTP, will be preferred.
+     *
+     * <p>
+     * {@code null} will be returned if no proxy was found in the environment.
+     *
+     * @param configuration The {@link Configuration} that is used to load proxy configurations from the environment. If
+     * {@code null} is passed then {@link Configuration#getGlobalConfiguration()} will be used. If {@link
+     * Configuration#NONE} is passed {@link IllegalArgumentException} will be thrown.
+     * @param resolveProxy Flag determining whether the returned {@link ProxyOptions} is resolved.
+     * @return A {@link ProxyOptions} reflecting a proxy loaded from the environment, if no proxy is found {@code null}
+     * will be returned.
+     * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE}.
+     */
+    public static ProxyOptions fromConfiguration(Configuration configuration, boolean resolveProxy) {
```

Comment:
Is this method required just for optimization to avoid hostname resolution? If the resolution fails, it will result in an unresolved address anyway.
https://docs.oracle.com/javase/7/docs/api/java/net/InetSocketAddress.html#InetSocketAddress(java.lang.String,%20int)

Reply:
The targeted reason for this method is to support resolving the proxy when a call is being made, and each time it is being made, instead of at construction time. There are cases where a consistent host name may change IP addresses, resulting in a fail request as the proxy was previously resolved to another address.